google/cloud/pubsub/internal/batching_publisher_connection_test.cc

```diff
@@ -64,9 +64,11 @@ TEST(BatchingPublisherConnectionTest, DefaultMakesProgress) {
   pubsub::Topic const topic("test-project", "test-topic");
 
   AsyncSequencer<void> async;
+  std::atomic<int> publish_count{0};
```

Comment:
Okay, pretty soon you will realize that I am dumb, because I always barely understand these synchronization primitives, and each time I learn enough to understand something, I forget in about 5 seconds flat.